src/ndpp_init.F90

```diff
@@ -0,0 +1,604 @@
+
+    integer :: i, j, k     ! loop indices
+    logical :: file_exists ! does ndpp_lib.xml exist?
+    integer :: filetype    ! default file type
+    logical :: nuscatter   ! is nuscatter data present?
+    logical :: chi_present ! is chi data present?
+    character(MAX_WORD_LEN)  :: directory   ! directory with cross sections
+    character(MAX_LINE_LEN)  :: temp_str
+    type(TallyObject), pointer :: t => null()
+    integer :: i_filter    ! Index of filter which contains energyin or energyout
+    ! We can use the same XSListing type for our ndpp data since the NDPP
+    ! data is a subset of whats in cross_sections.xml
+    type(XsListing), pointer :: listing => null()
+    type(Node), pointer      :: doc => null()
+    type(Node), pointer      :: node_ndpp => null()
+    type(NodeList), pointer  :: node_ndpp_list => null()
+    real(8), allocatable     :: ndpp_energy_bins(:)
+    integer                  :: order       ! ndpp_lib.xml's scattering order
+
+    ! Check if ndpp_lib.xml exists
+    inquire(FILE=ndpp_lib, EXIST=file_exists)
+    if (.not. file_exists) then
+      ! Could not find ndpp_lib.xml file
+      call fatal_error("NDPP Library XML file '" // trim(ndpp_lib) // &
+           "' does not exist!")
+    end if
+
+    call write_message("Reading NDPP Library XML file...", 5)
+
+    ! Parse ndpp_lib.xml file
+    call open_xmldoc(doc, ndpp_lib)
+
+    if (check_for_node(doc, "directory")) then
+      ! Copy directory information if present
+      call get_node_value(doc, "directory", directory)
+    else
+      ! If no directory is listed in ndpp_lib.xml, by default select the
+      ! directory in which the ndpp_lib.xml file resides
+      i = index(ndpp_lib, "/", BACK=.true.)
+      directory = ndpp_lib(1:i)
+    end if
+
+    ! determine whether binary, ascii, or hdf5
+    temp_str = ''
+    if (check_for_node(doc, "filetype")) &
+         call get_node_value(doc, "filetype", temp_str)
+    if (trim(temp_str) == 'ascii') then
+       filetype = ASCII
+    elseif (trim(temp_str) == 'binary') then
+       filetype = BINARY
+    elseif (trim(temp_str) == 'hdf5') then
+       filetype = H5
+    else
+       call fatal_error("Unknown filetype in " // trim(ndpp_lib) // &
+            ": " // trim(temp_str))
+    end if
+
+    ! Test metadata to ensure this library matches the problem definition
+    ! First test to see if we have a legendre scattering type and not tabular
+    ! (This will be removed at an undefined date when OpenMC supports
+    ! tabular scattering distributions as opposed to just Legendres.
+    ! Note that the authors have no immediate plans for this).
+    if (check_for_node(doc, "scatt_type")) &
+         call get_node_value(doc, "scatt_type", scatt_type)
+    if (scatt_type /= SCATT_TYPE_LEGENDRE) then
+      call fatal_error("Invalid Scattering Type represented in NDPP " // &
+           "data. Rerun NDPP with the Legendre scattering type set.")
```

Comment:
You don't need the `//` operator to concatenate these strings.  Just put an `&` at the end of each line (without closing the quotes) and another `&` at the beginning of the next line (without reopening the quotes).  Check out input_xml.F90 for examples of this.